### PR TITLE
added search functionality

### PR DESCRIPTION
**Pull request by ashishnnnnn**
-> User Can Click on The Searched Result and the new single product page will be opened.

Screenshot from 2022-04-08 12-02-08

-----------------------------------------------------------------------------------------------------------------------------------

Screenshot from 2022-04-08 12-02-12

-----------------------------------------------------------------------------------------------------------------------------------

Screenshot from 2022-04-08 12-02-27

### CODE REVIEW

**Review by SukanyaSen530 — APPROVED** [review]

**Review by junaidshaikh-js — APPROVED** [review]
Good work.

Liked the search implementation. Can be merged.

**Review by SukanyaSen530 — COMMENTED** [review]
You have merge conflicts, may sure to resolve it :)

File: src/utils/get_searched_result/get_searched_result.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const get_searched_result = (searchValue, product_list) => {
+  searchValue = searchValue.trim();
+  const searched_result = [];
+  if (searchValue.length === 0) {
+    return searched_result;
+  } else {
+    product_list.forEach((element) => {
```
Comment (tanishk15sharma) [inline src/utils/get_searched_result/get_searched_result.js:7]: You can use `.filter` rather than `forEach` it will improve your array immutability .

**Review by tanishk15sharma — APPROVED** [review]
Gave a suggestion , other things are good .

File: src/utils/get_searched_result/get_searched_result.js (hunk 1 above)
Comment (ashishnnnnn) [reply src/utils/get_searched_result/get_searched_result.js:7]: Actually i am generating every time a new array when this function is called..so i preferred this..

**Pull request merged**